packages/store/src/secure-store/index.ts

```diff
@@ -1,97 +1,102 @@
+import fs from 'fs';
 
-function getKeytarPaths(insiders: boolean): string[] {
+// Ensure the require is not bundled by webpack and resolved at runtime
+declare function __non_webpack_require__(moduleName: string): any;
+
+/**
+ * Retrieve paths to Zowe SDK from SAP Application Modeler extensions.
+ *
+ * @param insiders - Flag indicating if VS Code Insiders extensions should be checked.
+ * @returns Array of valid paths where Zowe SDK might exist.
+ */
+function getZoweSdkPaths(insiders: boolean): string[] {
     const vscodeRootPath = insiders ? '.vscode-insiders' : '.vscode';
-    const vscodeExtensionsPath = join(homedir(), vscodeRootPath, 'extensions/');
-    const AppMFoldersVscode =
-        fs
-            .readdirSync(vscodeExtensionsPath)
-            .filter((fn) => fn.startsWith('sapse.sap-ux-application-modeler-extension')) ?? [];
-    return AppMFoldersVscode.map((AppMFolderVscode) => {
-        const extensionPath = join(vscodeExtensionsPath, AppMFolderVscode);
-        const keytarPackageJsonPath = join(extensionPath, 'node_modules/keytar/package.json');
-        if (fs.existsSync(keytarPackageJsonPath)) {
-            return dirname(keytarPackageJsonPath);
-        } else {
-            return '';
-        }
-    }).filter((dirname) => dirname !== '');
+    const vscodeExtensionsPath = join(homedir(), vscodeRootPath, 'extensions');
+
+    if (!fs.existsSync(vscodeExtensionsPath)) {
+        return [];
+    }
+
+    const appModelerExtensions = fs
+        .readdirSync(vscodeExtensionsPath)
+        .filter((dir) => dir.startsWith('sapse.sap-ux-application-modeler-extension'));
+
+    return appModelerExtensions
+        .map((extensionDir) => {
+            const extensionPath = join(vscodeExtensionsPath, extensionDir);
+            const keytarPackageJsonPath = join(extensionPath, 'node_modules/@zowe/secrets-for-zowe-sdk/package.json');
+            return fs.existsSync(keytarPackageJsonPath) ? dirname(keytarPackageJsonPath) : '';
+        })
+        .filter((path) => path !== '');
 }
 
-function getKeytar(log: Logger): typeof Keytar | undefined {
+/**
+ * Load the Zowe secrets SDK, attempting fallbacks if the default import fails.
+ *
+ * @param log - Logger instance for logging.
+ * @returns The Zowe secrets SDK, or undefined if it cannot be loaded.
+ */
+function loadZoweSecretSdk(log: Logger): typeof zoweKeyring | undefined {
     try {
-        return require('keytar');
-    } catch (err) {
-        log.warn(errorString(err));
-        // Try to load keytar from sap-ux-application-modeler-extension node_modules if available this helps in some
-        // cases such as windows machines with restricted access. From node modules such as @sap/generator-fiori or @sap/ux-ui5-tooling
-        // keytar is not installed or is removed from the fs by virus scanner.
+        // Attempt to load the Zowe SDK directly
+        // eslint-disable-next-line
+        return require('@zowe/secrets-for-zowe-sdk').keyring;
```

Comment:
Hi @kjose90, what is the reason for a dynamic require here, why do we not use import, like
```typescript
import { keyring } from '@zowe/secrets-for-zowe-sdk';
```
?

Reply:
@Klaus-Keller we want to maintain the dynamic require like we had with keytar in case the binaries have been removed by virus scanner or could not be written on locked down PC's.